### ISSUE TITLE
Porting to other OS

### ISSUE THREAD

**Issue by RevoluPowered** [orbitersim/orbiter#131]
Would it be okay if I try to port this to Linux and Mac?

I would love to give it a shot.

P.S. Thanks for open sourcing, I hope that everything is okay! I saw the forum posts about it and you mentioned potentially taking down the orbiter website, but I don't think you need to.

**Comment by JDat** [comment]
You re not alone.
If I remember correctly, Martin (Author) have following milestones:
1) Release 32 bit bugfix version
2) Release 64 bit next gen version
3) Porting on other platforms

There is small discussion about porting (eat list first attempts to compile) on Linux: https://www.orbiter-forum.com/threads/building-openorbiter-under-linux.40122/

It would be great if you can port to linux, but not except that your work will be merged to main branch immediately. Maybe 1st thing to work for portability is to make crossplatform GUI stuff (QT widgets instead of native window GUI system)? And then will se how it's going.

**Comment by wiki-me** [comment]
In case anybody is interested,  someone is working on this [here](https://github.com/TheGondos/orbiter/tree/linux).

**Comment by EnjoMitch** [comment]
If I were to help here, I'd start piece by piece to replace all Windows calls to POSIX calls, and made sure that at each iteration:

- all tests pass
- my branches get merged

> It would be great if you can port to linux, but not except that your work will be merged to main branch immediately. Maybe 1st thing to work for portability is to make crossplatform GUI stuff (QT widgets instead of native window GUI system)?

This would be just one branch, that can be done independently of the simulator's bare bones. If you're trying to port the GUI, I recommend [wxWidgets](https://www.wxwidgets.org/) more, as it has proven to have a lot stabler API across the years and is much lighter that Qt.

**Comment by cederom** [comment]
Awsome! Please let me know when anything is ready to build and test on FreeBSD Unix :-)

**Comment by hardBSDk** [comment]
I suggest the [SDL](https://www.libsdl.org/) cross-platform framework, there are many options for all interfaces.

If you want something more easy, there's the [raylib](https://www.raylib.com/) library.

**Comment by cederom** [comment]
Yes SDL is portable everywhere works even on mobiles nowadays :-)

For GUI I recomment FLTK its simplest smallest fastest toolkit ever (much smaller and faster than GTK and QT and does not change that much) :-)
